src/python/pants/backend/project_info/tasks/export_dep_as_jar.py

```diff
@@ -191,6 +197,11 @@ def iter_transitive_jars(jar_lib):
         if isinstance(dep, Resources):
           resource_target_map[dep] = current_target
 
+      if hasattr(current_target, 'strict_deps'):
+        transitive_targets = DependencyContext.global_instance().dependencies_respecting_strict_deps(current_target)
+        for dep in transitive_targets:
+          info['transitive_targets'].append(dep.address.spec)
```

Comment:
1. Did a sample run with this on `dataproducts/aclservice/::`, the resulting json went from 4MB to 14MB, so would recommend putting this behind a flag. I haven't tried with a even larger project, but if json gets to 50MB, then that's likely not ideal.

2. Ideally we only need to put non-target-roots into `'transitive_targets`. Also relating to Borja's work to minimize the # of intellij modules. Do you think below whether below makes sense, @blorente?
For example:
```
Given
A -> B -> C -> D
A -> E -> F

$ ./pants export-dep-as-jar A
{
 A: {'transitive_targets', [B, C, D, E, F], is_target_root: True}
 B: // dependencies here doesn't matter because it's not target root, only need to show where the jar is
 C: // same
 D: // same
 E: // same
 F: // same
}

$ ./pants export-dep-as-jar A B
{ 
  A: { targets: [B], transitive_targets: [E,F], is_target_root: True}, // because B is also a target root
  B: { transitive_targets: [C, D], is_target_root: True}
  C: // dependencies here doesn't matter because it's not target root, only need to show where the jar is
  D: // same
  E: // same
  F: // same
```

Although the tricky part could be
```
A -> B -> C -> D
./pants export-dep-as-jar A C
```
In this case, we might have to force B into target root (intellij module rather than a library), because "3rdparty library" (B in this case from A's perspective) in intellij cannot depend back onto C.

To be clear, feel free to only do 1), and I'd be okay to ship this to unblock the work.

Reply:
So... now that you say that (apologies, should have noticed earlier), I think that what I was suggesting was related to https://github.com/pantsbuild/pants/pull/8736#issuecomment-559754334 : roughly:
1. Render the strict dep boolean
2. When a target is `strict_deps=True`, include the calculated _direct_ strict dependencies set in the dependencies list rather than the non-strict dependencies (alternatively define a second field `strict_dependencies=[...]`). When `strict_deps=False`, change nothing: the `dependencies` list would be as it is.

This would avoid the exponential blowup (because we'd never render transitive dependencies) and would avoid the strict deps calculation (which is the actual challenging one: unlike the non-strict deps calculation). Sorry, should have noticed earlier.

Reply:
Not sure if I follow on 2). For example A -> B , B -> C, D
```
scala_library(
   name = A
   strict_dep = True
   dependencies= [B]
)

scala_library(
   name = B
   strict_dep = True
   dependencies= [C, D]
   export=[C]
)

scala_library(
   name = C
   dependencies = []
)

scala_library(
   name = D
   dependencies = []
)
```
What would the output look like?
```
A: { 
    strict_dep=True, 
    dependencies = [B, C], // or should this be [B] only?
  }
B: {..}
C: {..}
```